condor_utils.py

```diff
@@ -721,3 +721,74 @@ def read_from_collector(address, history=False):
             logging.info('got %d entries', i)
         except Exception:
             logging.info('%s failed', schedd_ad['Name'], exc_info=True)
+
+def read_status_from_collector(address, after=datetime.now()-timedelta(hours=1)):
+    """Connect to condor collectors and schedds to pull job ads directly.
+
+    A generator that yields condor job dicts.
+
+    Args:
+        address (str): address of collector
+        history (bool): read history (True) or active queue (default: False)
+    """
+    import htcondor
+    coll = htcondor.Collector(address)
+    start_stamp = time.mktime(after.timetuple())
+    try:
+        gen = coll.query(
+            htcondor.AdTypes.Startd,
+            (
+                "EnteredCurrentState>={}"
+                .format(start_stamp)
```

Comment:
The idea here was not to weed out stale startds, but to select only those whose state has changed since the last run. It's true that this would miss startds running long jobs that are killed by the parent batch system between runs. I'll change this.